<?xml version="1.0"?>
<launch>
    <arg name="name" default="bebop"/>
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" />
    <arg name="use_sim_time" default="true"/>
    <arg name="world_name" default="basic"/>
    <arg name="csvFilesStoring" default="true"/>
    <arg name="csvFilesStoringTime" default="50.0"/> <!-- seconds -->
    <arg name="user_account" default="giuseppe"/> <!-- account name of your pc -->
    <arg name="waypoint_filter" default="true"/>
    <arg name="EKFActive" default="false"/>
    
  <env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find teamsannio_med_control)/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find teamsannio_med_control)/models"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find teamsannio_med_control)/worlds/$(arg world_name).world" />
  </include>

    <!-- Launch UAV -->
    <include file="$(find med_uav_description)/launch/spawn_bebop_only_odometry_and_camera.launch">
       <arg name="name" value="$(arg name)" />
       <arg name="x" value="$(arg x)"/>
       <arg name="y" value="$(arg y)"/>
       <arg name="z" value="$(arg z)"/>
    </include>

    <!-- Launch the controller  -->
    <node name="position_controller_node" pkg="teamsannio_med_control" type="position_controller_node" output="screen"> 
      <rosparam command="load" file="$(find teamsannio_med_control)/resource/controller_bebop.yaml" />
      <rosparam command="load" file="$(find teamsannio_med_control)/resource/bebop.yaml" />
      <rosparam command="load" file="$(find teamsannio_med_control)/resource/EKF_matrix.yaml" />
      <rosparam command="load" file="$(find teamsannio_med_control)/resource/waypoint_filter.yaml" />
      <param name="use_sim_time" value="$(arg use_sim_time)" />
      <param name="csvFilesStoring" value="$(arg csvFilesStoring)"/>
      <param name="csvFilesStoringTime" value="$(arg csvFilesStoringTime)"/>
      <param name="user_account" value="$(arg user_account)"/>
      <param name="EKFActive" value="$(arg EKFActive)"/>
      <param name="waypoint_filter" value="$(arg waypoint_filter)"/>
      <remap from="/command/motor_speed" to="/gazebo/command/motor_speed" />
      <remap from="/odometry" to="/bebop/odometry" />
      <remap from="/odometry_gt" to="/bebop/odometry_gt" />
      <remap from="/filteredOutput" to="/bebop/filteredOutput" />
      <remap from="/referenceAngles" to="/bebop/referenceAngles" />
      <remap from="/stateErrors" to="/bebop/stateErrors" />
      <remap from="/smoothedTrajectory" to="/bebop/smoothedTrajectory" />
      <remap from="/command/trajectory" to="/bebop/command/trajectory" />
      <remap from="/uTerrComponents" to="/bebop/uTerrComponents" />
      <remap from="/zVelocityComponents" to="/bebop/zVelocityComponents" />
      <remap from="/positionAndVelocityErrors" to="/bebop/positionAndVelocityErrors" />
      <remap from="/angularAndAngularVelocityErrors" to="/bebop/angularAndAngularVelocityErrors" />
    </node>

    <!-- Launch the trajectory generator -->
   <group ns="$(arg name)">
       <node name="waypoint_example" pkg="teamsannio_med_control" type="waypoint_example" output="screen" args="$(find teamsannio_med_control)/resource/waypoints.txt"/>   
    </group>

</launch>
